consensus/aura/aura.go

```diff
@@ -825,24 +826,25 @@ func (c *AuRa) Initialize(config *chain.Config, chain consensus.ChainHeaderReade
 
 }
 
-func (c *AuRa) ApplyRewards(header *types.Header, state *state.IntraBlockState, syscall consensus.SystemCall) error {
+func (c *AuRa) ApplyRewards(header *types.Header, state *state.IntraBlockState, syscall consensus.SystemCall, firehoseContext *firehose.Context) error {
 	beneficiaries, _, rewards, err := calculateRewards(c, header, syscall)
```

Comment:
calculate reward returns the reward kind list as second argument. You can use this to find the `addBalance` reason.

Something like this
```
                switch rewardKinds[i] {
		case aurainterfaces.RewardAuthor:
			state.AddBalance(beneficiaries[i], rewards[i], false, firehoseContext, firehose.BalanceChangeReason("reward_mine_block"))
		case aurainterfaces.RewardUncle:
			state.AddBalance(beneficiaries[i], rewards[i], false, firehoseContext, firehose.BalanceChangeReason("reward_mine_uncle"))
		default:
			state.AddBalance(beneficiaries[i], rewards[i], false, firehoseContext, firehose.BalanceChangeReason("reward_unknown"))
		}
```